### ISSUE TITLE
Feature Request: .ics URLs for Events and Birthdays

### ISSUE THREAD

**Issue by mritzmann** [monicahq/monica#2891]
**Is your feature request related to a problem?**
Monica already has WebDAV Support. However, i would be happy if Monica would also offer `.ics`-support.

My reasons:

- Some clients have a better ics integration instead of webdav (e.g. many web clients)
- .ics is easier to set up than WebDav (ics only needs a secret URL, no separate credentials)
- .ics has read-only access by default, that's enough for me

**Describe the solution you'd like**

- New setting under: `/settings/ical` (similar to `/settings/dav`)
- A url for events
- A url for birthdays
- Optional: A third URL with both
- Url should be secret, not guessable (maybe with username and api key in url)

**Comment by silva96** [comment]
subscribed to this must-have feature!

**Comment by asbiin** [comment]
We already have an ics export at:
https://[hostname]/dav/calendars/[email]/birthdays?export
The only problem is it need credentials, so we have to add some token for instance to protect it.

Also I don't now what information to add for Reminders?

**Comment by mritzmann** [comment]
Thanks @asbiin for your answer. The background for me is that I want to replace every reminder email with it. (So I won't get any more e-mails. Whereby birthdays are more important to me).

> Also I don't now what information to add for Reminders?

Good question. I think that the date on which the entry is made should be the same as the date on which the reminder email is sent.

**Comment by mritzmann** [comment]
I've been rethinking that. I used to write this:

> I think that the date on which the entry is made should be the same as the date on which the reminder email is sent.

Now I think that's wrong. In a calendar you look into the future from time to time for planning. Thus newly summarized:

- **For the reminder ical feed:** Then when the event takes place.
- **For the birthday ical feed:** Then when the person has birthday.

So i think that there is no need for a logic. The two ical feeds simply have to map 1:1 what is in Monica.

**Comment by tborychowski** [comment]
It would also make it easier to integrate with e.g. roundube's carddav.
Currently the issue is that the password that's allowed there is `varchar(255)`, and tokens generated in monicahq are ~1000 chars long.
I modified the DB manually, so that I was able to integrate both, but it's hacky, and I'd prefer a native solution :-)

**Comment by jackeymai** [comment]
> I've been rethinking that. I used to write this:
> 
> > I think that the date on which the entry is made should be the same as the date on which the reminder email is sent.
> 
> Now I think that's wrong. In a calendar you look into the future from time to time for planning. Thus newly summarized:
> 
> * **For the reminder ical feed:** Then when the event takes place.
> * **For the birthday ical feed:** Then when the person has birthday.
> 
> So i think that there is no need for a logic. The two ical feeds simply have to map 1:1 what is in Monica.

I also want this feature, it's more easy to manage your daily life.

**Comment by Dinth** [comment]
+1 for this, since i moved to Monica im constantly forgetting about people bdays :(

**Comment by dalekirkwood** [comment]
Another plus 1 here. I can't seen to get WebDAV to work with Thunderbird due to token length and also google doesn't support authenticated WebDAV.

**Comment by Dinth** [comment]
Proton Calendar doesnt support authenticated DAV either. 

It would be nice to be able to use a secret url instead without need of authentication (similarly to how it does both Google and Microsoft) https://proton.me/support/subscribe-to-external-calendar#get-external-link

**Comment by Dinth** [comment]
Or even leave everything as it is, just adding an option to disable authentication on DAV items - both the likeliness and the severity of the threat are already quite low.

**Comment by mrUlrik** [comment]
@Dinth I completely agree, I have the exact same use case. I'm giving up Nextcloud because it's a bit overkill and have switched to Proton.

Unrelated tangent:
> As an aside, I'd been looking for a Card/CalDAV replacement for ages. I just accidentally stumbled across Monica and thought "Oh cool, this would be a cool middle ground." Then I noticed it had Card/CalDAV support and I immediately fell in love.

I want Monica to be my "source of truth" in all aspects of contact management. I'd love to have things like Proton Calendar be able to subscribe to Monica birthday and task calendars.

**Comment by Dinth** [comment]
@mrUlrik were you able to sync Monica calendar with NextCloud? It doesnt seem to support authentication either.

**Comment by mrUlrik** [comment]
Sorry! I hadn't tried, but my experience with Nextcloud says it probably could not. By the time I found Monica I'd already started dismantling Nextcloud.
